api/v2alpha1/pipeline_spec.proto

```diff
@@ -665,6 +665,8 @@ message ArtifactTypeSchema {
 message PipelineTaskInfo {
   // The display name of the task.
   string name = 1;
+  // The name of the task used for the driver/executor

```

Comment:
So this name is unique throughout the pipeline yes? we should mention that here as this is part of API. We can remove mentions of driver/executor as that's backend implementation detail.